### PR TITLE
Add support for maintenance window on google_container_cluster

### LINKED ISSUE #526: Add support for maintenance windows on google_container_cluster
Hi there,

Google Cloud Container Engine maintenance windows were introduced last week:

https://cloud.google.com/container-engine/release-notes#new-features_1
https://cloud.google.com/container-engine/docs/maintenance-window

As of now, the relevant API is just made of `MaintenancePolicy` with a nested `DailyMaintenanceWindow` (probably making room future alternative policies):

https://github.com/google/google-api-go-client/blob/cf8a9e7cedca7843a0fdd8adc9d019e84349768d/container/v1/container-gen.go#L1255

Not yet documented at https://cloud.google.com/container-engine/reference/rest/v1beta1/projects.locations.clusters

### Affected Resource

- google_container_cluster

### Terraform Configuration Files

I'd suggest the following design:

```hcl
resource "google_container_cluster" "primary" {
  name               = "marcellus-wallace"
  zone               = "us-central1-a"
  initial_node_count = 3

  maintenance_policy {
    daily_maintenance_window {
       start_time = "3:00"
    }
  }

  master_auth {
    username = "mr.yoda"
    password = "adoy.rm"
  }

  node_config {
    oauth_scopes = [
      "https://www.googleapis.com/auth/compute",
    ]
}
```

### PR DESCRIPTION
**Pull request by michaelbannister**
Fixes #526 

This is my first venture into both the Go language and the internals of Terraform. Please be forthcoming with feedback but don't be too harsh if I've done something dumb! ;)

I have run my own standalone test creating a cluster with a maintenance window set, but I haven't yet run the acceptance tests, because the readme says it costs money but gives no indication of _how much_ money that might be!

### CODE REVIEW

**Comment by michaelbannister** [comment]
@danawillow I can merge master to my branch or would you prefer I rebase it? (I'm aware of the dangers of doing this if someone has already cloned it.)

**Comment by danawillow** [comment]
I'm fine with either!

**Comment by michaelbannister** [comment]
@danawillow I've rebased and look forward to any feedback you or others may have.

File: google/resource_container_cluster.go [hunk 1]
```diff
@@ -172,6 +172,36 @@ func resourceContainerCluster() *schema.Resource {
...
+			"maintenance_policy": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"daily_maintenance_window": {
+							Type:     schema.TypeList,
+							Required: true,
```
Comment (danawillow) [inline google/resource_container_cluster.go]: Just for the sake of future-proofing, let's go ahead and `ForceNew: true` this one too

File: google/resource_container_cluster_test.go [hunk 2]
```diff
@@ -576,6 +576,25 @@ func TestAccContainerCluster_withNodePoolNodeConfig(t *testing.T) {
...
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckContainerClusterDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccContainerCluster_withMaintenanceWindow("03:00"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckContainerCluster(
+						"google_container_cluster.primary"),
```
Comment (danawillow) [inline google/resource_container_cluster_test.go]: this should be "google_container_cluster.with_maintenance_window" to match the resource name in the test config

File: website/docs/r/container_cluster.html.markdown [hunk 3]
```diff
@@ -243,6 +260,10 @@ exported:
 * `instance_group_urls` - List of instance group URLs which have been assigned
     to the cluster.
 
+* `maintenance_policy.daily_maintenance_window.duration` - Duration of the time window, automatically chosen to be
```
Comment (danawillow) [inline website/docs/r/container_cluster.html.markdown]: Nested fields are fun! This is actually `maintenance_policy.0.daily_maintenance_window.0.duration`

Comment (michaelbannister) [reply website/docs/r/container_cluster.html.markdown]: Oh yes - I see this was recently updated for the master_auth stuff (this explains why it didn't work when I tried to use those attributes without the 0 index recently!)

**Comment by michaelbannister** [comment]
Thanks, Dana!

**Comment by danawillow** [comment]
You're welcome, thanks for the quick turnaround!

**Comment by pdecat** [comment]
Hi, thanks for that!

Just came across an issue though:

```
      maintenance_policy.#:                                       "1" => "1"                 
      maintenance_policy.0.daily_maintenance_window.#:            "1" => "1"                                                                                                              
      maintenance_policy.0.daily_maintenance_window.0.duration:   "PT4H0M0S" => <computed>                                                                                                
      maintenance_policy.0.daily_maintenance_window.0.start_time: "6:00" => "06:00" (forces new resource)                                                                                 
```

Will open an issue.

**Pull request merged**